src/transforms/bin.js

```diff
@@ -65,9 +65,12 @@ export function bin(outputs = {fill: "count"}, options = {}) {
 }
 
 function maybeDenseInterval(bin, k, options = {}) {
-  return options?.interval == null
-    ? options
-    : bin({[k]: options?.reduce === undefined ? reduceFirst : options.reduce, filter: null}, options);
+  if (options?.interval == null) return options;
+  const reduce = options?.reduce === undefined ? reduceFirst : options.reduce;
```

Comment:
Since `options?.interval` is non-nullish here, `options` is guaranteed to be non-nullish, and we can remove the optional chaining. Hence, this can be written as:

```suggestion
  const {reduce = reduceFirst} = options;
```